docs/DesignDocs/Attestation_API_Proposal.md

```diff
@@ -0,0 +1,534 @@
+Proposal of OE SDK Attestation Public and Plugin API for V0.10 Release
+====
+
+# Background
+
+This document describes the proposed OE SDK attestation public and plugin API
+for V0.10 release, as an evolution of the experimental attestation API
+implemented in the
+[OE SDK V0.9 release](https://github.com/openenclave/openenclave/tree/v0.9.x).
+
+This proposal is based on these efforts:
+- Design document [Attestation V3 Update]:
+[Custom Attestation Data Formats for Open Enclave V3 Updates](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/CustomAttestation_V3.md).
+- Discussion in [Issue #2729]: [Outstanding issues on attestation plugin design](https://github.com/openenclave/openenclave/issues/2729).
+- Design document [quote-ex Integration]:
+[Attestation: OE SDK Integration with Intel® SGX SDK quote-ex Library for Generation of Evidence in New Formats](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/SGX_QuoteEx_Integration.md).
+- Design document [Design Notes]:
+[Notes on OE SDK Attestation API and Plugin Library Design](https://github.com/shnwc/openenclave/blob/master/docs/DesignDocs/NotesOnAttestationAPI.md).
+  - Note: this document is in the process of being up-streamed as OE SDK [PR #2801](https://github.com/openenclave/openenclave/pull/2801).
+
+# OE SDK Attestation Public API Proposal
+
+## Existing OE SDK V0.9 Public API
+
+The OE SDK V0.9 experimental public API for attestation is composed of
+functions for a number of use cases to be implemented by applications.
+The use cases fall into two categories: evidence generation and verification.
```

Comment:
As I mention in a later comment, I think there should be 3 categories: Attester, Verifier, and Relying Party.   See https://tools.ietf.org/html/draft-ietf-rats-architecture (of which I would point out that Intel is a co-author and WG chair)

Reply:
My understanding is that the Relying Party role in the RATS model is played by verifier applications, and thus not in the scope of the public API, but a user of the API to consume evidence verification result. I'll add some notes for clarification.

Reply:
Verifier applications and Relying Party applications are two different things, not to be confused.

Reply:
Understood. Application developers can use the OE SDK attestation API to build Attester and Verifier roles / applications. This API does not provide functions to build Relying Party application to convert evidence verification results into signed Attestation Result.

Reply:
Today this API supports applications that are combined verifier+relying party. In the long term relying part and verifier could be different. Non blocker.

Reply:
Added the text between lines 40 - 44.